src/EFCore.PG.NodaTime/Query/ExpressionTranslators/Internal/NpgsqlNodaTimeMemberTranslatorPlugin.cs

```diff
@@ -76,9 +76,49 @@ public NpgsqlNodaTimeMemberTranslator(ISqlExpressionFactory sqlExpressionFactory
                 return TranslateDateTime(instance, member, returnType);
             }
 
+            if (instance is not null && declaringType == typeof(Duration))
+            {
+                return TranslateDuration(instance, member);
+            }
+
             return null;
         }
 
+        /// <summary>
+        /// Translates Duration members
+        /// </summary>
+        /// <param name="instance"></param>
+        /// <param name="member"></param>
+        /// <returns>The translated expression or null</returns>
+        private SqlExpression? TranslateDuration(SqlExpression instance, MemberInfo member)
+        {
+            switch (member.Name)
```

Comment:
Nit: can make this into a switch expression, making the whole method expression-bodied. This would mean either duplicating the code for TotalDays/Hours/Minutes/Seconds, or extracting a tiny local method (would also remove the need for a second internal switch).

Reply:
done